### PR TITLE
AVP Pipelines

### PR DESCRIPTION
**Pull request by tarrencev**
Adding support for pipelines on the AVP. These can be either set on by default or toggled by the user through an API. Right now, a pipeline is just a set of parallel processors. Eventually, I'd like to extend it to support a dynamic processing DAG. This lays the ground work for that.

Each processor implements the `Element` interface and the `Pipeline` is a collection of elements. The sample builder is always on by default and writes `Sample`s to the pipeline.

The pipeline is fully configurable in `cmd/avp/main.go`. So it is simple for others to implement their own elements and integrate them with the AVP processing pipeline.

I ported over some rtc code from the base libraries. In particular, `pipeline` is similar to `router` but operates on `Sample` rather than `rtp.Packet`s. Now they are pretty much decoupled. So these changes shouldn't affect anything in the stable path.

Any feedback would be appreciated!

### CODE REVIEW

File: pkg/node/sfu/internal.go [hunk 1]
```diff
@@ -148,7 +148,6 @@ func unpublish(msg map[string]interface{}) (map[string]interface{}, *nprotoo.Err
 	mid := util.Val(msg, "mid")
 	router := rtc.GetOrNewRouter(mid)
 	if router != nil {
-		router.Close()
```
Comment (tarrencev) [inline pkg/node/sfu/internal.go:151]: This is called in `rtc.DelRouter` below

**Review by adwpc — COMMENTED** [review]
delete cmd/.DS_Store?

**Review by adwpc — APPROVED** [review]
LGTM

**Review by jbrady42 — APPROVED** [review]
Looks like a good start. Some of this could probably start to get extracted into own repo like webmsaver.

**Comment by tarrencev** [comment]
Good idea. I think we can extract the elements to a separate `ion-avp-elements` repo. We can keep all processing modules there. That will allow us to avoid dependency bloat here.

**Pull request merged**